components/automate-gateway/gateway/middleware/authv2/authv2.go

```diff
@@ -83,13 +83,44 @@ func (c *client) Handle(ctx context.Context, subjects []string, projectsToFilter
 	return auth_context.NewContext(ctx, subjects, projects, resource, action, middleware.AuthV2.String()), nil
 }
 
+type resp struct {
+	ctx        context.Context
+	authorized bool
+}
+
+
+func (r *resp) Ctx() context.Context {
+	return r.ctx
+}
+
+func (r *resp) GetAuthorized() bool {
+	return r.authorized
+}
+
 func (c *client) IsAuthorized(ctx context.Context, subjects []string, resource, action string,
-) (middleware.AuthorizationResponse, error) {
-	return c.client.IsAuthorized(ctx, &authz.IsAuthorizedReq{
-		Subjects: subjects,
-		Resource: resource,
-		Action:   action,
+	projectsToFilter []string) (middleware.AuthorizationResponse, error) {
+	log := ctxlogrus.Extract(ctx)
+	filteredResp, err := c.client.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
+		Subjects:       subjects,
+		Resource:       resource,
+		Action:         action,
+		ProjectsFilter: projectsToFilter,
```

Comment:
This is looking similar to the `Handle` function. When do we hit `Handle` vs `IsAuthorized`?

Reply:
Is it? [`Handle`](https://github.com/chef/automate/blob/a2a442f16459f887d517bc43447a154d4b729b3b/components/automate-gateway/gateway/middleware/authv2/authv2.go#L26-L84) is to be used in a GRPC interceptor, and depends on 
a) the grpc method invoked, and
b) the policy info being mapped in our proto files
i.e. [this half of the method](https://github.com/chef/automate/blob/a2a442f16459f887d517bc43447a154d4b729b3b/components/automate-gateway/gateway/middleware/authv2/authv2.go#L26-L55), neither of which exist for custom handlers.

I'll outline how the authorizer is supposed to work in another comment, give me a minute 😉